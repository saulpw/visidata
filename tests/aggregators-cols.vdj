#!vd -p
{"sheet": "global", "col": null, "row": "disp_date_fmt", "longname": "set-option", "input": "%b %d, %Y", "keystrokes": "", "comment": null}
{"longname": "open-file", "input": "sample_data/test.jsonl", "keystrokes": "o"}
{"sheet": "test", "col": "key2", "row": "", "longname": "key-col", "input": "", "keystrokes": "!", "comment": "toggle current column as a key column"}
{"sheet": "test", "col": "key2", "row": "", "longname": "addcol-aggregate", "input": "count", "comment": "add column(s) with aggregator of rows grouped by key columns"}
{"sheet": "test", "col": "qty", "row": "", "longname": "type-float", "input": "", "keystrokes": "%", "comment": "set type of current column to float"}
{"sheet": "test", "col": "qty", "row": "", "longname": "addcol-aggregate", "input": "rank sum", "comment": "add column(s) with aggregator of rows grouped by key columns"}
{"sheet": "test", "col": "qty_sum", "row": "", "longname": "addcol-sheetrank", "input": "", "comment": "add column with the rank of each row based on its key columns"}
