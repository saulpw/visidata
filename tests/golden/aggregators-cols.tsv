key2	key2_count	key1	qty	qty_rank	qty_sum	test_sheetrank	amt	
foo	2	2016-01-01 11:00:00	1.00	1	31.00	5		
	0	2016-01-01 1:00	2.00	1	66.00	2	3	
baz	3		4.00	1	292.00	4	43.2	
#ERR	0	#ERR	#ERR	1	0.00	1	#ERR	#ERR
bar	2	2017-12-25 8:44	16.00	2	16.00	3	.3	
baz	3		32.00	2	292.00	4	3.3	
	0	2018-07-27 4:44	64.00	2	66.00	2	9.1	
bar	2	2018-07-27 16:44		1	16.00	3		
baz	3	2018-07-27 18:44	256.00	3	292.00	4	.01	
foo	2	2018-10-20 18:44	30.00	2	31.00	5	.01	
